cloudpickle/cloudpickle.py

```diff
@@ -69,6 +69,43 @@
     from io import BytesIO as StringIO
     PY3 = True
 
+
+try:
+    from ctypes import pythonapi, py_object, c_int, PYFUNCTYPE
+except ImportError:
+    supports_recursive_closure = False
+
+    def compress_closure(closure):
+        return closure
+
+    def decompress_closure(compressed_closure):
+        return compressed_closure
+
+    def fill_cells(cells, values):
+        pass
+else:
+    supports_recursive_closure = True
+
+    def compress_closure(closure):
```

Comment:
Ok, I understand better now. Can you add a comment explaining design choices at the beginning of those changes? So that further readers don't get lost.